src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/ObjectExplorerService.cs

```diff
@@ -689,6 +705,204 @@ private static void AddToApplicableChildFactories(Dictionary<string, HashSet<Chi
+                if (parentNames != null)
+                {
+                    clonedParentNames = parentNames.ToList();
+                }
+                TreeFolder lastFolder = null;
+                var currentTreeFolder = treeFolder;
+                var path = new StringBuilder(name);
+                if (schema != null)
+                {
+                    path.Insert(0, schema + ".");
+                }
+
+                while (currentTreeFolder != null)
+                {
+                    if (currentTreeFolder.Name == "Server" || (currentTreeFolder.Name == "Databases" && oeSession.Root.NodeType == "Database"))
+                    {
+                        var serverRoot = oeSession.Root;
+                        if (oeSession.Root.NodeType == "Database")
+                        {
+                            serverRoot = oeSession.Root.Parent;
+                            path.Insert(0, oeSession.Root.NodeValue + (path.Length > 0 ? "/" : ""));
+                        }
+
+                        path.Insert(0, serverRoot.NodeValue + (path.Length > 0 ? "/" : ""));
+                        break;
+                    }
+
+                    if (lastFolder == null || currentTreeFolder.ChildFolders.Contains(lastFolder))
+                    {
+                        path.Insert(0, currentTreeFolder.LocLabel + "/");
+                        lastFolder = currentTreeFolder;
+                        currentTreeFolder = currentTreeFolder.ParentFolder;
+                    }
+                    else
+                    {
+                        if (currentTreeFolder.ContainedType == "Database")
+                        {
+                            path.Insert(0, databaseName + "/");
+                        }
+                        else if (clonedParentNames.Count > 0)
+                        {
+                            var parentName = clonedParentNames.Last();
+                            clonedParentNames.RemoveAt(clonedParentNames.Count - 1);
+                            path.Insert(0, parentName + "/");
+                        }
+                        else
+                        {
+                            path = null;
+                            break;
+                        }
+
+                        lastFolder = null;
+                    }
+                }
+
+                if (path != null)
+                {
+                    outputPaths.Add(path.ToString());
+                }
+            }
+
+            var nodes = new List<TreeNode>();
+            foreach (var outputPath in outputPaths)
+            {
+                var treeNode = oeSession.Root.FindNodeByPath(outputPath, true);
+                if (treeNode != null)
+                {
+                    nodes.Add(treeNode);
+                }
+            }
+            return nodes;
+        }
+
+        private void LoadTreeStructure()
+        {
+            var assembly = typeof(ObjectExplorerService).Assembly;

```

Comment:
catch exception and log the error, if for some reason the resource cannot be loaded, the exception will cause the service to fail